plugin/etcd/setup.go

```diff
@@ -99,8 +99,9 @@ func etcdParse(c *caddy.Controller) (*Etcd, error) {
 
 func newEtcdClient(endpoints []string, cc *tls.Config, username, password string) (*etcdcv3.Client, error) {
 	etcdCfg := etcdcv3.Config{
-		Endpoints: endpoints,
-		TLS:       cc,
+		Endpoints:         endpoints,
+		TLS:               cc,
+		DialKeepAliveTime: etcdTimeout + etcdTimeout/2,
```

Comment:
@journey-c Sorry for the late response. Is there any reference in why this value is chosen?

Reply:
@yongtang The value is based on the operation timeout and buffer time, in order to timely remove faulty nodes when the ETCD network is not working

Reply:
@journey-c  So that is the empirical configuration from your working environment?

Reply:
@yongtang Yes, but does require "DialKeepAliveTime"

Reply:
@journey-c  My concern is that "etcdTimeout + etcdTimeout/2" is a magic number. This will be harder to maintain in the future (as people may not know why `1+ 1/2` is the right number. - What about just use `DialKeepAliveTime`? (not `1+1/2`)

Reply:
@yongtang It should be possible too